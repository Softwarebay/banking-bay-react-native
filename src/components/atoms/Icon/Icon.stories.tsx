import { color, number } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { colors } from 'theme';
import { ViewStory } from 'utils/storybook/ViewStory';

import { Icon } from './Icon';

storiesOf('Atoms/Icon', module)
	.addDecorator((getStory) => (
		<ViewStory style={{ backgroundColor: colors.screenBackground }} center>
			{getStory()}
		</ViewStory>
	))
	.add('Amazon', () => <Icon name="amazon" size={number('size', 20)} />)
	.add('Check', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="check"
			size={number('size', 20)}
		/>
	))
	.add('Chevron Left', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="chevron-left"
			size={number('size', 20)}
		/>
	))
	.add('Credit Card', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="credit-card"
			size={number('size', 20)}
		/>
	))
	.add('Credit Cards', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="credit-cards"
			size={number('size', 20)}
		/>
	))
	.add('Deposits', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="deposits"
			size={number('size', 20)}
		/>
	))
	.add('Dollar Sign', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="dollar-sign"
			size={number('size', 20)}
		/>
	))
	.add('Eye Off', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="eye-off"
			size={number('size', 20)}
		/>
	))
	.add('Facebook', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="facebook"
			size={number('size', 20)}
		/>
	))
	.add('FlagUSA', () => <Icon name="flag-usa" size={number('size', 20)} />)
	.add('Google +', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="google-plus"
			size={number('size', 20)}
		/>
	))
	.add('Wallet', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="loans"
			size={number('size', 20)}
		/>
	))
	.add('Logo', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="logo"
			size={number('size', 20)}
		/>
	))
	.add('Money Transfer', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="money-transfer"
			size={number('size', 20)}
		/>
	))
	.add('More', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="more"
			size={number('size', 20)}
		/>
	))
	.add('Notifications', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="notifications"
			size={number('size', 20)}
		/>
	))
	.add('Paypal', () => <Icon name="paypal" size={number('size', 20)} />)
	.add('Profile', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="profile"
			size={number('size', 20)}
		/>
	))
	.add('Repeat', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="repeat"
			size={number('size', 20)}
		/>
	))
	.add('Dashboard', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="dashboard"
			size={number('size', 20)}
		/>
	))
	.add('Smartphone', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="smartphone"
			size={number('size', 20)}
		/>
	))
	.add('Twitter', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="twitter"
			size={number('size', 20)}
		/>
	))
	.add('Search', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="search"
			size={number('size', 20)}
		/>
	))
	.add('Home', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="home"
			size={number('size', 20)}
		/>
	))
	.add('ATM', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="atm"
			size={number('size', 20)}
		/>
	))
	.add('eBay', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="ebay"
			size={number('size', 20)}
		/>
	))
	.add('Info', () => (
		<Icon
			color={color('color', colors.mainDark)}
			name="info"
			size={number('size', 20)}
		/>
	));
