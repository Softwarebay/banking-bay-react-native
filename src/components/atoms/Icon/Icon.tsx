import {
	Amazon,
	Atm,
	Check,
	ChevronLeft,
	CreditCard,
	CreditCards,
	Deposit,
	DollarSign,
	Ebay,
	EyeOff,
	Facebook,
	FlagUSA,
	GooglePlus,
	Home,
	Info,
	LoanWallet,
	Logo,
	MoneyTransfer,
	MoreTiles,
	NotificationBell,
	Paypal,
	ProfileIcon,
	Repeat,
	Report,
	Search,
	Smartphone,
	Twitter,
} from 'assets/icons';

export interface Props {
	name:
		| 'facebook'
		| 'twitter'
		| 'check'
		| 'eye-off'
		| 'google-plus'
		| 'logo'
		| 'chevron-left'
		| 'credit-card'
		| 'dollar-sign'
		| 'repeat'
		| 'smartphone'
		| 'profile'
		| 'credit-cards'
		| 'deposits'
		| 'loans'
		| 'notifications'
		| 'dashboard'
		| 'more'
		| 'amazon'
		| 'money-transfer'
		| 'paypal'
		| 'flag-usa'
		| 'search'
		| 'atm'
		| 'ebay'
		| 'home'
		| 'info';
	size: number;
	color?: string;
}

// eslint-disable-next-line consistent-return
export const Icon = ({ name, color, size }: Props) => {
	switch (name) {
		case 'facebook':
			return (
				<Facebook
					color={color}
					height={size}
					testID="facebook-icon-test-id"
					width={size}
				/>
			);
		case 'google-plus':
			return (
				<GooglePlus
					color={color}
					height={size}
					testID="google-plus-icon-test-id"
					width={size}
				/>
			);
		case 'twitter':
			return (
				<Twitter
					color={color}
					height={size}
					testID="twitter-icon-test-id"
					width={size}
				/>
			);
		case 'logo':
			return (
				<Logo
					color={color}
					height={size}
					testID="logo-icon-test-id"
					width={size}
				/>
			);
		case 'check':
			return (
				<Check
					color={color}
					height={size}
					testID="check-icon-test-id"
					width={size}
				/>
			);
		case 'eye-off':
			return (
				<EyeOff
					color={color}
					height={size}
					testID="eye-off-icon-test-id"
					width={size}
				/>
			);
		case 'chevron-left':
			return (
				<ChevronLeft
					color={color}
					height={size}
					testID="chevron-left-icon-test-id"
					width={size}
				/>
			);
		case 'credit-card':
			return (
				<CreditCard
					color={color}
					height={size}
					testID="credit-card-icon-test-id"
					width={size}
				/>
			);
		case 'smartphone':
			return (
				<Smartphone
					color={color}
					height={size}
					testID="smartphone-icon-test-id"
					width={size}
				/>
			);
		case 'dollar-sign':
			return (
				<DollarSign
					color={color}
					height={size}
					testID="dollar-sign-icon-test-id"
					width={size}
				/>
			);
		case 'repeat':
			return (
				<Repeat
					color={color}
					height={size}
					testID="repeat-icon-test-id"
					width={size}
				/>
			);
		case 'profile':
			return (
				<ProfileIcon
					color={color}
					height={size}
					testID="profile-icon-test-id"
					width={size}
				/>
			);
		case 'credit-cards':
			return (
				<CreditCards
					color={color}
					height={size}
					testID="credit-cards-icon-test-id"
					width={size}
				/>
			);
		case 'deposits':
			return (
				<Deposit
					color={color}
					height={size}
					testID="deposits-icon-test-id"
					width={size}
				/>
			);
		case 'more':
			return (
				<MoreTiles
					color={color}
					height={size}
					testID="more-icon-test-id"
					width={size}
				/>
			);
		case 'dashboard':
			return (
				<Report
					color={color}
					height={size}
					testID="dashboard-icon-test-id"
					width={size}
				/>
			);
		case 'notifications':
			return (
				<NotificationBell
					color={color}
					height={size}
					testID="notifications-icon-test-id"
					width={size}
				/>
			);
		case 'loans':
			return (
				<LoanWallet
					color={color}
					height={size}
					testID="loans-icon-test-id"
					width={size}
				/>
			);
		case 'paypal':
			return (
				<Paypal
					color={color}
					height={size}
					testID="paypal-icon-test-id"
					width={size}
				/>
			);
		case 'amazon':
			return (
				<Amazon
					color={color}
					height={size}
					testID="amazon-icon-test-id"
					width={size}
				/>
			);
		case 'money-transfer':
			return (
				<MoneyTransfer
					color={color}
					height={size}
					testID="money-transfer-icon-test-id"
					width={size}
				/>
			);
		case 'flag-usa':
			return (
				<FlagUSA
					color={color}
					height={size}
					testID="flag-usa-icon-test-id"
					width={size}
				/>
			);
		case 'search':
			return (
				<Search
					color={color}
					height={size}
					testID="search-icon-test-id"
					width={size}
				/>
			);
		case 'home':
			return (
				<Home
					color={color}
					height={size}
					testID="home-icon-test-id"
					width={size}
				/>
			);
		case 'atm':
			return (
				<Atm
					color={color}
					height={size}
					testID="atm-icon-test-id"
					width={size}
				/>
			);
		case 'ebay':
			return (
				<Ebay
					color={color}
					height={size}
					testID="ebay-icon-test-id"
					width={size}
				/>
			);
		case 'info':
			return (
				<Info
					color={color}
					height={size}
					testID="info-icon-test-id"
					width={size}
				/>
			);
		// no default
	}
};
