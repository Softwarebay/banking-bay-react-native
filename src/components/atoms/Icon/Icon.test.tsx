import { render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { Icon } from './Icon';

import type { Props } from './Icon';

describe('Icon', () => {
	const props: Omit<Props, 'name'> = {
		size: 20,
		color: 'red',
	};

	it('should match snapshot - facebook icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="facebook" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - twitter icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="twitter" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - check icon', () => {
		const tree = testRenderer.create(<Icon {...props} name="check" />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - eye-off icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="eye-off" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - google-plus icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="google-plus" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - logo icon', () => {
		const tree = testRenderer.create(<Icon {...props} name="logo" />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - chevron-left icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="chevron-left" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - credit-card icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="credit-card" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - dollar-sign icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="dollar-sign" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - repeat icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="repeat" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - smartphone icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="smartphone" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - profile icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="profile" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - credit-cards icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="credit-cards" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - deposits icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="deposits" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - loans icon', () => {
		const tree = testRenderer.create(<Icon {...props} name="loans" />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - notifications icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="notifications" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - dashboard icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="dashboard" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - more icon', () => {
		const tree = testRenderer.create(<Icon {...props} name="more" />).toJSON();

		expect(tree).toMatchSnapshot();
	});
	it('should match snapshot - amazon icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="amazon" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - money-transfer icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="money-transfer" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - paypal icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="paypal" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - flag-usa icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="flag-usa" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - search icon', () => {
		const tree = testRenderer
			.create(<Icon {...props} name="search" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render facebook icon', () => {
		render(<Icon {...props} name="facebook" />);

		const icon = screen.getByTestId('facebook-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render twitter icon', () => {
		render(<Icon {...props} name="twitter" />);

		const icon = screen.getByTestId('twitter-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render check icon', () => {
		render(<Icon {...props} name="check" />);

		const icon = screen.getByTestId('check-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render eye-off icon', () => {
		render(<Icon {...props} name="eye-off" />);

		const icon = screen.getByTestId('eye-off-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render google-plus icon', () => {
		render(<Icon {...props} name="google-plus" />);

		const icon = screen.getByTestId('google-plus-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render logo icon', () => {
		render(<Icon {...props} name="logo" />);

		const icon = screen.getByTestId('logo-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render chevron-left icon', () => {
		render(<Icon {...props} name="chevron-left" />);

		const icon = screen.getByTestId('chevron-left-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render credit-card icon', () => {
		render(<Icon {...props} name="credit-card" />);

		const icon = screen.getByTestId('credit-card-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render dollar-sign icon', () => {
		render(<Icon {...props} name="dollar-sign" />);

		const icon = screen.getByTestId('dollar-sign-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render repeat icon', () => {
		render(<Icon {...props} name="repeat" />);

		const icon = screen.getByTestId('repeat-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render smartphone icon', () => {
		render(<Icon {...props} name="smartphone" />);

		const icon = screen.getByTestId('smartphone-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render profile icon', () => {
		render(<Icon {...props} name="profile" />);

		const icon = screen.getByTestId('profile-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render credit-cards icon', () => {
		render(<Icon {...props} name="credit-cards" />);

		const icon = screen.getByTestId('credit-cards-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render deposits icon', () => {
		render(<Icon {...props} name="deposits" />);

		const icon = screen.getByTestId('deposits-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render loans icon', () => {
		render(<Icon {...props} name="loans" />);

		const icon = screen.getByTestId('loans-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render notifications icon', () => {
		render(<Icon {...props} name="notifications" />);

		const icon = screen.getByTestId('notifications-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render dashboard icon', () => {
		render(<Icon {...props} name="dashboard" />);

		const icon = screen.getByTestId('dashboard-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render more icon', () => {
		render(<Icon {...props} name="more" />);

		const icon = screen.getByTestId('more-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render amazon icon', () => {
		render(<Icon {...props} name="amazon" />);

		const icon = screen.getByTestId('amazon-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render money-transfer icon', () => {
		render(<Icon {...props} name="money-transfer" />);

		const icon = screen.getByTestId('money-transfer-icon-test-id');

		expect(icon).not.toBeNull();
	});

	it('should render paypal icon', () => {
		render(<Icon {...props} name="paypal" />);

		const icon = screen.getByTestId('paypal-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render flag-usa icon', () => {
		render(<Icon {...props} name="flag-usa" />);

		const icon = screen.getByTestId('flag-usa-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});

	it('should render search icon', () => {
		render(<Icon {...props} name="search" />);

		const icon = screen.getByTestId('search-icon-test-id');

		expect(icon).not.toBeNull();

		expect(icon.props).toHaveProperty('height', 20);
		expect(icon.props).toHaveProperty('width', 20);
		expect(icon.props).toHaveProperty('color', 'red');
	});
});
