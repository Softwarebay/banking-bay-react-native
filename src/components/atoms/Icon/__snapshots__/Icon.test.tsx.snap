// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should match snapshot - ATM icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="atm-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - amazon icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="amazon-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - check icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="check-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - chevron-left icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="chevron-left-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - credit-card icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="credit-card-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - credit-cards icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="credit-cards-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - dashboard icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="dashboard-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - deposits icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="deposits-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - dollar-sign icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="dollar-sign-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - eBay icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="ebay-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - eye-off icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="eye-off-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - facebook icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="facebook-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - flag-usa icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="flag-usa-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - google-plus icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="google-plus-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - home icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="home-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - info icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="info-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - loans icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="loans-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - logo icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="logo-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - money-transfer icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="money-transfer-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - more icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="more-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - notifications icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="notifications-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - paypal icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="paypal-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - profile icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="profile-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - repeat icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="repeat-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - search icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="search-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - smartphone icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="smartphone-icon-test-id"
  width={20}
/>
`;

exports[`Icon should match snapshot - twitter icon 1`] = `
<SvgMock
  color="red"
  height={20}
  testID="twitter-icon-test-id"
  width={20}
/>
`;
